contracts/src/arbitration/dispute-kits/DisputeKitClassic.sol

```diff
@@ -285,13 +286,11 @@ contract DisputeKitClassic is BaseDisputeKit, IEvidence {
             core.getCurrentPeriod(_coreDisputeID) == KlerosCore.Period.commit,
             "The dispute should be in Commit period."
         );
-        require(_commit != bytes32(0), "Empty commit.");
```

Comment:
We don't really need to remove this to allow changing the commitment. There is no valid case for allowing a 0x00 commitment, it would likely be a bug or a human error so it seems like good UX to have it.

Reply:
Ah sorry I forgot to explain this change.

I think most of users will cast the commit through a UI. So the UI can do the validations. I though would be better to remove that validation from the contracts, as it looks unlikely to happen and now the user can override the vote commitment later in case of submitting a zero one by mistake.

But if you disagree with this reasoning I will push a change reverting that validation removal.

Reply:
I think it's better to keep the require to avoid some flaky behaviour in case someone wants to bypass the UI. Also the user might submit empty commit by mistake without noticing

Reply:
Okay let's not change how data validation is done in this particular PR. Data validation could be reviewed separately across the board in a separate PR with a proper discussion of UX vs less code/gas.

Reply:
Makes sense to treat it separately if needed. I pushed [the change](https://github.com/kleros/kleros-v2/pull/190/commits/233e7d33e3e0f89390a9571cd1e34917b2f21d0f) to keep the validation!